open_feature/hooks/hook_support.py

```diff
@@ -27,7 +27,7 @@ def after_all_hooks(
     flag_type: FlagType,
     hook_context: HookContext,
     hooks: typing.List[Hook],
-    hints: dict,
+    hints: typing.Optional[typing.Mapping],
```

Comment:
Yes. As per `Conditional Requirement 4.2.2.1` the Hook Context must be immutable.
`dict` is a `MutableMapping`, using `Mapping` ensures that `mypy` will yell at you if you try to do 
`hints['abc'] = 123`

We should probably set default arguments for optional values, IMHO, yes.

Reply:
@matthewelwell  Ah, I didn't change the typing of the hints in all places. Done so now.

Reply:
Glad to see there are types to support it and there will be linting errors internally for any invalid mutations :rocket:

My PR in #31 wraps this up as a runtime error as well, since not all Python projects necessarily check or care about type.